### ISSUE TITLE
Rework explicit partitioning + Cass RP/Murmur3 error handling

### ISSUE THREAD

**Issue by dalaro** [thinkaurelius/titan#706]
Titan on Cassandra-Murmur3 allows definition of partitioned vertex labels, but then throws a cryptic precondition failure when attempting to allocate an ID for vertices with that label:

```
gremlin> g = TitanFactory.open('conf/titan-cassandra-es.properties')
==>titangraph[cassandrathrift:[127.0.0.1]]                                                                                                                                                   
gremlin> mgmt = g.getManagementSystem()                                                                                                                                                      
==>com.thinkaurelius.titan.graphdb.database.management.ManagementSystem@12f14707
gremlin> mgmt.makeVertexLabel('user').make()
==>user
gremlin> mgmt.makeVertexLabel('product').partition().make()
==>product
gremlin> mgmt.commit()
==>null
gremlin> v = g.addVertex("product")
1
Display stack trace? [yN] y
java.lang.IllegalArgumentException: 1
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:93)
        at com.thinkaurelius.titan.graphdb.database.idassigner.VertexIDAssigner.assignID(VertexIDAssigner.java:299)
        at com.thinkaurelius.titan.graphdb.database.idassigner.VertexIDAssigner.assignID(VertexIDAssigner.java:185)
        at com.thinkaurelius.titan.graphdb.database.idassigner.VertexIDAssigner.assignID(VertexIDAssigner.java:156)
        at com.thinkaurelius.titan.graphdb.database.StandardTitanGraph.assignID(StandardTitanGraph.java:344)
        at com.thinkaurelius.titan.graphdb.transaction.StandardTitanTx.addVertex(StandardTitanTx.java:440)
        at com.thinkaurelius.titan.graphdb.transaction.StandardTitanTx.addVertex(StandardTitanTx.java:464)
        at com.thinkaurelius.titan.graphdb.transaction.StandardTitanTx.addVertex(StandardTitanTx.java:459)
        at com.thinkaurelius.titan.graphdb.blueprints.TitanBlueprintsTransaction.addVertex(TitanBlueprintsTransaction.java:78)
        at com.thinkaurelius.titan.graphdb.blueprints.TitanBlueprintsGraph.addVertex(TitanBlueprintsGraph.java:155)
        at com.thinkaurelius.titan.graphdb.blueprints.TitanBlueprintsGraph.addVertex(TitanBlueprintsGraph.java:27)
```

Can we check for this in `partition()` or `mgmt.commit()`, and just prevent the label from being defined?  If not, then we should at least detect when explicit partitioning is combined with RP/Murmur3 and throw a more descriptive exception.

**Comment by mbroecheler** [comment]
We should check the following:

1) when building a partitioned vertex label, it should be verified that
cluster.partition is true.

2) when initializing a graph, it should be checked that the storage backend
is keyordered if cluster.partition is true. The later must already be the
case, if not please add.

**Comment by dalaro** [comment]
1. Makes sense, added precondition and test coverage in 1e5e54bf5cfad4aa4c4540887c9ab8f07e933d1f.  Could we add another check in `mgmt.makeVertexLabel("foo").partition().make()` to catch this earlier?  Or would that break something?
2. We're already doing this, though the check is `isDistributed && isKeyOrdered` rather than just `isKeyOrdered`.  I tweaked the log statements emitted during this check as part of 1e5e54bf5cfad4aa4c4540887c9ab8f07e933d1f and added a precondition in there, but the config behavior is unchanged.

**Comment by dalaro** [comment]
6c3914ad754ddd53d8118a6350dc6587ec1f6639 fixes up some tests that I missed in 024c66755e8fd6187944895a6681a5c95b4260ce and 628d48fe23431422e71225306ce2220611a45c86.

**Comment by dalaro** [comment]
I moved the cluster.partition check into `StandardVertexLabelMaker.make`, removed the `addVertex` check introduced in 1e5e54bf5cfad4aa4c4540887c9ab8f07e933d1f, and updated tests to expect `make` on partitioned vertex labels to fail when `cluster.partition` is false (and check that they can be created when it is true).  I think this is set.
